deploy/src/crd.rs

```diff
@@ -0,0 +1,291 @@
+        Serialize, Toleration,
+    };
+
+    /// Xline cluster specification
+    #[derive(CustomResource, Deserialize, Serialize, Clone, Debug, JsonSchema)]
+    #[cfg_attr(test, derive(Validate))]
+    #[cfg_attr(test, garde(context(ClusterSpec)))]
+    #[kube(
+        group = "xlineoperator.datenlord.io",
+        version = "v1",
+        kind = "XlineCluster",
+        singular = "xlinecluster",
+        plural = "xlineclusters",
+        struct = "Cluster",
+        namespaced,
+        status = "ClusterStatus",
+        shortname = "xc",
+        scale = r#"{"specReplicasPath":".spec.size", "statusReplicasPath":".status.size"}"#,
+        printcolumn = r#"{"name":"CronSpec", "type":"string", "description":"The cron spec defining the interval a backup CronJob is run", "jsonPath":".spec.backup.cron"}"#,
+        printcolumn = r#"{"name":"Size", "type":"string", "description":"The cluster size", "jsonPath":".spec.size"}"#,
+        printcolumn = r#"{"name":"StorageType", "type":"string", "description":"The backup storage type", "jsonPath":".spec.backup.storage_type"}"#
+    )]
+    pub(crate) struct ClusterSpec {
+        /// Size of the xline cluster, less than 3 is not allowed
+        #[cfg_attr(test, garde(range(min = 3)))]
+        #[schemars(range(min = 3))]
+        pub(crate) size: usize,
+        /// Pod specification
+        #[cfg_attr(test, garde(dive))]
+        pub(crate) pod: PodSpec,
+        /// Backup specification
+        #[cfg_attr(test, garde(custom(option_backup_dive)))]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub(crate) backup: Option<BackupSpec>,
+    }
+
+    /// Xline cluster backup specification
+    #[derive(Serialize, Deserialize, Clone, Debug, JsonSchema)]
+    #[cfg_attr(test, derive(Validate))]
+    pub(crate) struct BackupSpec {
+        /// Cron Spec
+        #[cfg_attr(test, garde(pattern(r"^(?:\*|[0-5]?\d)(?:[-/,]?(?:\*|[0-5]?\d))*(?: +(?:\*|1?[0-9]|2[0-3])(?:[-/,]?(?:\*|1?[0-9]|2[0-3]))*){4}$")))]
+        #[schemars(regex(
+            pattern = r"^(?:\*|[0-5]?\d)(?:[-/,]?(?:\*|[0-5]?\d))*(?: +(?:\*|1?[0-9]|2[0-3])(?:[-/,]?(?:\*|1?[0-9]|2[0-3]))*){4}$"
+        ))]
+        pub(crate) cron: String,
+        /// Backup storage type, one of ["s3", "pv"]
+        #[cfg_attr(test, garde(skip))]
+        #[serde(flatten)]
+        pub(crate) storage: StorageSpec,
+    }
+
+    /// Xline cluster backup storage specification
+    #[derive(Serialize, Deserialize, Clone, Debug, JsonSchema)]
+    #[serde(untagged)]
+    pub(crate) enum StorageSpec {
+        #[serde(rename = "s3")]
+        S3 { s3: S3Spec },
+        #[serde(rename = "pv")]
+        PV { pv: String },
+    }
+
+    /// Xline cluster backup S3 specification
+    #[derive(Serialize, Deserialize, Clone, Debug, JsonSchema)]
+    pub(crate) struct S3Spec {
+        /// S3 bucket path to use for backup
+        /// TODO validate
```

Comment:
Can we validate it in this pr? Or we'd better to leave an issue to track it.

Reply:
I'll fix it in #8